test/integration_test/utils/px_operator.go

```diff
@@ -60,3 +63,60 @@ func getPXOperatorImageTag() (string, error) {
 
 	return strings.Split(image, ":")[1], nil
 }
+
+// GetPxOperatorImage return PX Operator image
+func GetPxOperatorImage() (string, error) {
+	deployment, err := appops.Instance().GetDeployment(PortworxOperatorDeploymentName, PxNamespace)
+	if err != nil {
+		return "", err
+	}
+
+	var image string
+	for _, container := range deployment.Spec.Template.Spec.Containers {
+		if container.Name == PortworxOperatorContainerName {
+			image = container.Image
+			logrus.Infof("Get portworx-operator image installed: %s", image)
+			break
+		}
+	}
+
+	return image, nil
+}
+
+// GetPxOperatorDeployment return PX Operator deployment
+func GetPxOperatorDeployment() (*appsv1.Deployment, error) {
+	pxOperator, err := appops.Instance().GetDeployment(PortworxOperatorDeploymentName, PxNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	return pxOperator, nil
+}
+
+// UpdateAndValidatePxOperator update and validate PX Operator deployment
+func UpdateAndValidatePxOperator(pxOperator *appsv1.Deployment, f func(*appsv1.Deployment) *appsv1.Deployment, t *testing.T) *appsv1.Deployment {
+	livePxOperator, err := appops.Instance().GetDeployment(pxOperator.Name, pxOperator.Namespace)
+	require.NoError(t, err)
+
+	newPxOperator := f(livePxOperator)
+
+	latestLivePxOperator, err := UpdatePxOperator(newPxOperator)
+	require.NoError(t, err)
+
+	err = appops.Instance().ValidateDeployment(latestLivePxOperator, DefaultValidateUpdateTimeout, DefaultValidateUpdateRetryInterval)
+	require.NoError(t, err)
+
+	return latestLivePxOperator
+}
+
+// UpdatePxOperator update PX Operator deploymnent
+func UpdatePxOperator(pxOperator *appsv1.Deployment) (*appsv1.Deployment, error) {
+	logrus.Infof("Update PX Operator %s in %s", pxOperator.Name, pxOperator.Namespace)
+
+	pxOperator, err := appops.Instance().UpdateDeployment(pxOperator)
+	if err != nil {
+		return nil, err
+	}
+
+	return pxOperator, nil
```

Comment:
same here - you could just write `return appops.Instance().UpdateDeployment(pxOperator)`, if you want.

Reply:
Done